helix-core/src/main/java/org/apache/helix/model/CloudConfig.java

```diff
@@ -207,6 +228,16 @@ public Builder(CloudConfig cloudConfig) {
       _cloudInfoProcessorName = cloudConfig.getCloudInfoProcessorName();
     }
 
+    public Builder(String clusterName, boolean enabled, CloudProvider cloudProvider, String cloudID,
+        List<String> cloudInfoSource, String cloudInfoProcessorName) {
+      _clusterName = clusterName;
```

Comment:
As we discussed offline, the clustername here is just used for the ZNRecord Id. This brings some concerns.
1. The admin API will have the cluster  name as an input, then another cluster name passed through the cloud config. This is duplicate and potentially conflict information.
2. The ZNRecord id itself will be the same as the cluster config node id. Which is not good.

Reply:
Fixed. Cluster name has been removed.